src/capwap_http_api_handler.erl

```diff
@@ -0,0 +1,378 @@
+%% Copyright (C) 2013-2017, Travelping GmbH <info@travelping.com>
+
+%% This program is free software: you can redistribute it and/or modify
+%% it under the terms of the GNU Affero General Public License as published by
+%% the Free Software Foundation, either version 3 of the License, or
+%% (at your option) any later version.
+
+%% This program is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%% GNU Affero General Public License for more details.
+
+%% You should have received a copy of the GNU Affero General Public License
+%% along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+-module(capwap_http_api_handler).
+
+-include("../include/capwap_packet.hrl").
+-include("../include/capwap_config.hrl").
+-include("../include/capwap_ac.hrl").
+
+-export([start_link/0]).
+-export([init/2, content_types_provided/2,
+         handle_request_json/2,
+         allowed_methods/2,
+         content_types_accepted/2]).
+
+-define(DEFAULT_PORT,	8000).
+-define(DEFAULT_IP,     {127, 0, 0, 1}).
+-define(ACCEPTORS_NUM,  100).
+
+-record(s, {
+    verbose = false :: boolean(),
+    opts
+}).
+
+start_link() ->
+    HttpConfig = application:get_env(capwap, http_api, []),
+    Port = proplists:get_value(port, HttpConfig, ?DEFAULT_PORT),
+    IP = proplists:get_value(ip, HttpConfig, ?DEFAULT_IP),
+    INet = get_inet(IP),
+    AcceptorsNum = proplists:get_value(acceptors_num, HttpConfig, ?ACCEPTORS_NUM),
+    Dispatch = cowboy_router:compile([
+		{'_', [
+			{"/api/v1/version",                          ?MODULE, []},
+
+			{"/api/v1/capwap/list",                      ?MODULE, []},
+			{"/api/v1/capwap/show/:id",                  ?MODULE, []},
+			{"/api/v1/capwap/update/:id/:link/:hash",    ?MODULE, []},
+			{"/api/v1/capwap/set_ssid/:id/:ssid[/:rid]", ?MODULE, []},
+			{"/api/v1/capwap/stop_radio/:id/:rid",       ?MODULE, []},
+
+			{"/api/v1/station/list",                     ?MODULE, []},
+			{"/api/v1/station/detach/:id",               ?MODULE, []},
+
+			{"/api/v1/dp/wtp_list",                      ?MODULE, []},
+			{"/api/v1/dp/stats",                         ?MODULE, []}
+		]}
+	]),
+    TransOpts = [{port, Port}, {ip, IP}, INet, {num_acceptors, AcceptorsNum}],
+    ProtoOpts = #{env => #{dispatch => Dispatch}},
+    cowboy:start_clear(capwap_http_api, TransOpts, ProtoOpts).
+
+get_inet({_, _, _, _}) ->
+    inet;
+get_inet({_, _, _, _, _, _, _, _}) ->
+    inet6.
+
+init(Req, Opts) ->
+    Verbose = cowboy_req:header(<<"verbose">>, Req, <<"false">>),
+    {cowboy_rest, Req, #s{verbose = (Verbose == <<"true">>), opts = Opts}}.
+
+allowed_methods(Req, State) ->
+    {[<<"GET">>], Req, State}.
```

Comment:
`GET` only is kind of "wrong" in the REST world when you want to trigger something like "stop radio".

Reply:
may be DELETE for stop_radio and PUSH for update, set-ssid?